src/templates/generated/action.py.njk

```diff
@@ -46,10 +46,16 @@ class {{action.actionName}}(argparse.{{action.baseClass}}):
             {% if action.type != 'dictionary' or action.subProperties.length > 0 %}
             {% for subprop in action.subProperties %}
             {% if loop.first %}if{% else %}elif{% endif %} kl == '{{subprop.nameAz}}':
+                sub_d = d
+                {% for parentKey in subprop.parentKeys %}
+                if '{{parentKey}}' not in sub_d:
+                    sub_d['{{parentKey}}'] = {}
+                sub_d = sub_d['{{parentKey}}']
+                {% endfor %}
                 {% if subprop.type == 'array' %}
-                d['{{subprop.namePython}}'] = v
+                sub_d['{{subprop.namePython}}'] = v
                 {% else %}
-                d['{{subprop.namePython}}'] = v[0]
+                sub_d['{{subprop.namePython}}'] = v[0]
```

Comment:
based on the error log https://dev.azure.com/azure-sdk/public/_build/results?buildId=951502&view=logs&j=6d695271-00fc-56c3-4bc2-e5786380b36e&t=4281d9bd-7c10-5dfd-5545-77cf0693a38f&l=3159 

d has never been reassigned after d = {}  sub_d has been constantly been assigned as d. I don't think we can get the right result. what you  want is something like  d[parent_key] = sub_d  ? 

In fact, we also need to consider cases like when there'are multiple parent layers  such as d['parent_key1']['parent_key2']'''['parent_keyn'] =  v[0], one sub_d may not be able to express such cases.

we also need to consider whether d['parent_key1'] to d['parent_key1']...['parent_keyn-1']  exists.

Reply:
d['parent_key1']['parent_key2']'''['parent_keyn'] = v[0] already concerned.

Reply:
I will show you a demo of generated code tomorrow.

Reply:
Below is the example of generated code. You can run it in python console.

```python
properties = {"id": ["this is the id"], "user-assigned-identity": ["this is user assigned identity"]}
d = {}
for k in properties:
    kl = k.lower()
    v = properties[k]
    if kl == 'id':
        sub_d = d
        sub_d['id'] = v[0]
    elif kl == 'user-assigned-identity':
        sub_d = d
        if 'parentKey1' not in sub_d:
            sub_d['parentKey1'] = {}
        sub_d = sub_d['parentKey1']
        if 'parentKey2' not in sub_d:
            sub_d['parentKey2'] = {}
        sub_d = sub_d['parentKey2']
        sub_d['user_assigned_identity'] = v[0]

print(d)
```

The result is

```json
{"id": 'this is the id', "parentKey1": {"parentKey2": {'user_assigned_identity': 'this is user assigned identity'}}}
```

Reply:
The reason this code can work is because as d is a dict, sub_d = d only done a pointer copy instead of deep copy. which is a trick usage of Python language. Not sure that should be in standard CLI code. May be difficult for trouble shooting.

Reply:
Python is always using pointer copy except basic types. It can be in standard CLI code.

Reply:
Then please fix the scenario test. Just update the baseline

Reply:
No problem. I'm still fixing another issue related to action. I will fix tests after that.